tests/Integration/Database/QueryBuilderTest.php

```diff
@@ -167,4 +168,65 @@ public function testPaginateWithSpecificColumns()
             (object) ['title' => 'Bar Post', 'content' => 'Lorem Ipsum.'],
         ]);
     }
+
+    public function testWhereWithApply()
+    {
+        $result = DB::table('posts')->apply(function (Builder $builder) {
+            $builder->where('title', 'Foo Post');
+        })->get();
+
+        $this->assertCount(1, $result);
+        $this->assertSame('Foo Post', $result->first()->title);
+        $this->assertSame('2017-11-12 13:14:15', $result->first()->created_at);
+    }
+
+    public function testGroupByWithApply()
+    {
+        $result = DB::table('posts')->apply(function (Builder $builder) {
+            $builder->groupBy('content');
+        })->get();
+
+        $this->assertCount(1, $result);
+    }
+
+    public function testApplyInvokableScope()
+    {
+        $result = DB::table('posts')->apply(new PostFrom2017Scope)->get();
```

Comment:
I think **CreatedIn2017Scope** is more suitable? or What about this?

```php
$result = DB::table('posts')->apply((new CreatedInYearScope(2017)))->get();
```

Reply:
I don't think that test class is relevante